pkg/ec2metadata/ec2metadata.go

```diff
@@ -196,16 +196,17 @@ func (e *Service) GetRebalanceRecommendationEvent() (rebalanceRec *RebalanceReco
 // GetMetadataInfo generic function for retrieving ec2 metadata
 func (e *Service) GetMetadataInfo(path string) (info string, err error) {
 	resp, err := e.Request(path)
-	if resp != nil && (resp.StatusCode < 200 || resp.StatusCode >= 300) {
-		return "", fmt.Errorf("Metadata request received http status code: %d", resp.StatusCode)
-	}
 	if err != nil {
 		return "", fmt.Errorf("Unable to parse metadata response: %w", err)
 	}
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", fmt.Errorf("Unable to parse http response: %w", err)
+		return "", fmt.Errorf("Unable to parse http response. Status code: %d. %w", resp.StatusCode, err)
+	}
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		log.Debug().Msgf("Metadata response status code: %d. Body: %s", resp.StatusCode, string(body))
```

Comment:
Any thoughts on always printing StatusCode and body for log-level debug and for these non-success status codes always print for log-level info/warning? My thinking is the non-success statuses can impact how NTH runs (i.e. terminate if errThreshold is hit) so that information should be more readily accessible.

Reply:
Non-success status codes will already be in the logs, regardless of log level, because they are in the error message.

We don't yet know how useful (or big) the body will be, so I'm more inclined to introduce it at log-level debug so that it won't affect existing customers who don't want or need that extra information. If it proves useful, though, we should definitely log it at info level instead.

Logging success status codes and their bodies is probably a bit superfluous, because we can assume those requests from their downstream effects (i.e., if you're querying for an instance ID and you get it successfully, we will know it was successful because the caller did not fail out and log something).